validator/sawtooth_validator/execution/scheduler.py

```diff
@@ -52,8 +52,24 @@ def get_batch_execution_result(self, batch_signature):
         raise NotImplementedError()
 
     @abstractmethod
+    def get_transaction_execution_results(self, batch_signature):
+        """Get all TransactionExecutionResults for a batch. If the batch isn't
+        finished or is invalid, returns None.
```

Comment:
Even if the Batch is invalid, wouldn't we want to return a list of TransactionExecutionResults, to see which one failed? I would expect only when the batch is incomplete would we want to return None.